query.go

```diff
@@ -763,6 +763,21 @@ checkResponses:
 			// fixed before exposing this function for public use.
 			checkResponse(sm.sp, sm.msg, queryQuit)
 
+			// Each time we receive a response from the current
+			// peer, we'll reset the main peer timeout as they're
+			// being responsive.
+			if !peerTimeout.Stop() {
+				<-peerTimeout.C
+			}
+			peerTimeout.Reset(qo.timeout)
+
+			// Also at this point, if the peerConnectTimeout is
+			// still active, then we can disable it, as we're
+			// receiving responses from the current peer.
+			if connectionTimeout != nil && !connectionTimeout.Stop() {
+				<-connectionTimeout.C
```

Comment:
This will hang on the second time it is called (in the second message).

Reply:
Have you seen this happen? Or is this a premonition? 

Reworked slightly to rely on the `nil` check as was originally intended.

Reply:
From the godoc:
> Stop prevents the Timer from firing. It returns true if the call stops the timer, false if the timer has already expired or been stopped.

Looks like it would indeed hang because false is returned on the second `Stop` call, but this second call won't happen in the latest diff due to setting the ticker to `nil` below.